File: back/src/main/java/com/back/domain/wish/entity/Wish.java
package com.back.domain.wish.entity;

import com.back.domain.book.entity.Book;
import com.back.domain.member.entity.Member;
import com.back.global.jpa.entity.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
public class Wish extends BaseEntity {

    @ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name = "member_id")
    private Member member;

    @ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name = "book_id")
    private Book book;

    public Wish(Member member, Book book) {
        this.member = member;
        this.book = book;
    }

}

File: back/src/main/java/com/back/domain/wish/entity/Wish.kt
package com.back.domain.wish.entity

import com.back.domain.book.entity.Book
import com.back.domain.member.entity.Member
import com.back.global.jpa.entity.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
class Wish(
    @field:JoinColumn(name = "member_id")
    @field:ManyToOne(fetch = FetchType.LAZY)
    val member: Member,

    @field:JoinColumn(name = "book_id")
    @field:ManyToOne(fetch = FetchType.LAZY)
    val book: Book,
) : BaseEntity()
